pkg/controllers/node/node_syncer.go

```diff
@@ -67,12 +67,17 @@ func (c *NodeController) OnUpdates(updates []bapi.Update) {
 				c.syncNodeLabels(obj.(*v1.Node))
 			}
 		case bapi.UpdateTypeKVDeleted:
-			n := upd.KVPair.Value.(*apiv3.Node)
-			if kn := getK8sNodeName(*n); kn != "" {
+			if upd.KVPair.Value != nil {
+				logrus.Warnf("KVPair value should be nil for Deleted UpdataType")
+			}
+
+			// Try to perform unmapping based on resource name.
+			n := upd.KVPair.Key.(model.ResourceKey).Name
+			if n != "" {
 				// Remove it from the node map.
-				logrus.Debugf("Unmapping Calico -> k8s node. %s -> %s", n.Name, kn)
+				logrus.Debugf("Unmapping Calico -> k8s node. %s -> %s", n, n)
 				c.nodemapLock.Lock()
-				delete(c.nodemapper, kn)
+				delete(c.nodemapper, n)
```

Comment:
I don't think this is quite right, since the node mapper is a map of Kubernetes node name to Calico node name, and we're now deleting the Calico node name as they key (the two might be different).

I think we may need to iterate through the map and delete the entry with a value that matches the Calico node, _or_ keep a reverse mapping as well.

Reply:
I see. I am not have clear idea of why we have two names for the node. When does Calico node name differ with Kubernetes node name?

I will make changes accordingly.

Reply:
The Calico node name is often based on `os.Hostname()` whereas the Kubernetes node name might be set differently.

For KDD mode we enforce them to be the same, but in etcd mode they can be set separately.